services/cloudapi/cloudapinic/nic.go

```diff
@@ -0,0 +1,270 @@
+package cloudapinic
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	ionoscloud "github.com/ionos-cloud/sdk-go/v6"
+	"github.com/ionos-cloud/terraform-provider-ionoscloud/v6/services/cloudapi"
+	"github.com/ionos-cloud/terraform-provider-ionoscloud/v6/utils"
+)
+
+type Service struct {
+	Client *ionoscloud.APIClient
+	Meta   interface{}
+	D      *schema.ResourceData
+}
+
+func (fs *Service) Get(ctx context.Context, datacenterID, serverID string, depth int32) ([]ionoscloud.Nic, error) {
+	nics, apiResponse, err := fs.Client.NetworkInterfacesApi.DatacentersServersNicsGet(ctx, datacenterID, serverID).Depth(depth).Execute()
+	apiResponse.LogInfo()
+	if err != nil {
+		return nil, err
+	}
+	if apiResponse.HttpNotFound() || nics.Items == nil || len(*nics.Items) == 0 {
+		log.Printf("[DEBUG] no nics found for datacenter %s, server %s", datacenterID, serverID)
```

Comment:
@cristiGuranIonos not sure about this: what happens if the server doesn't exist? I suppose that we will receive an `http not found` response here, so then the message would be confusing. Can you please test this scenario?

Reply:
this needs to be removed actually. It never reaches here since httpnotfound also throws an error above and returns